### PR TITLE
prov/efa: support 0 byte read

### PR DESCRIPTION
**Pull request by wzamazon**
fix an issue that 0 byte read is not allowed.

### CODE REVIEW

File: prov/efa/src/rxr/rxr_read.c [hunk 1]
```diff
@@ -553,69 +633,33 @@ int rxr_read_post(struct rxr_ep *ep, struct rxr_read_entry *read_entry)
...
-			msg.addr = read_entry->addr;
-			self_comm = (read_entry->context_type == RXR_READ_CONTEXT_PKT_ENTRY);
-			if (self_comm)
-				pkt_entry->flags |= RXR_PKT_ENTRY_LOCAL_READ;
-			ret = efa_rma_post_read(efa_ep, &msg, 0, self_comm);
-		}
-
-		if (OFI_UNLIKELY(ret)) {
-			rxr_pkt_entry_release_tx(ep, pkt_entry);
+			read_once_len = MIN(read_once_len, rxr_env.efa_read_segment_size);
```
Comment (zhngaj) [inline prov/efa/src/rxr/rxr_read.c:650]: This is not related to what this PR really does. Just wondering if we can use `max_read_size` here, as L603 already calculate the max read size based on lower ep type.

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_read.c:650]: Sure. I think it is good idea. added that in new revision

**Review by zhngaj — APPROVED** [review]
